examples/helloworld/greeter_server/main.go

```diff
@@ -55,11 +60,25 @@ func (s *server) SayHello(ctx context.Context, in *pb.HelloRequest) (*pb.HelloRe
 }
 
 func main() {
-	lis, err := net.Listen("tcp", port)
-	if err != nil {
-		log.Fatalf("failed to listen: %v", err)
-	}
+	flag.Parse()
+
 	s := grpc.NewServer()
 	pb.RegisterGreeterServer(s, &server{})
-	s.Serve(lis)
+
+	log.Printf("Running hello server on %s ...", *listen)
+	if *httpMode {
+		// Running a gRPC server when you need to integrate with an existing
+		// net/http server on the same port: (using HTTP routing as needed)
+		http.Handle("/", s)
```

Comment:
Ah, turns out this is because `http2.ConfigureServer` mutates `server.TLSConfig`, which is not the config we're using to create our listener. Initializing our `server.TLSConfig` to our config before the call to `ConfigureServer` makes this work!